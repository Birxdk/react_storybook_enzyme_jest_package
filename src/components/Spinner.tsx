import * as React from 'react';
import styled from 'styled-components';
import {
  black
} from '../core/colors';

const SpinningDiv = styled.div`
  transform-origin: center center;
  animation: spin 1s infinite;

  @keyframes spin {
    100% {
      transform: rotate(360deg);
    }
  }
`;

const Container = styled.div`
  height: 100%;
  width: 100%;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;

  .spinner-children {
    display: block;
    color: ${black};
  }
`;

const Spinner: React.SFC = ({ children }) => {
  console.log(children);
  return (
    <Container>
      <SpinningDiv>
        {children &&
          <div className="spinner-children">
            {children}
          </div>}
      </SpinningDiv>
    </Container>
  );
};
export default Spinner;
