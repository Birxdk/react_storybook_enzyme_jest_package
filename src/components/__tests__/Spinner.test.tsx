import { shallow } from 'enzyme';
import * as React from 'react';
import 'jest-styled-components';
import * as renderer from 'react-test-renderer';
import Spinner from '../Spinner';

describe('Wait spinner', () => {
  it('renders the spinner without children', () => {
    const component = renderer.create(<Spinner />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the spinner with children', () => {
    const component = renderer.create(<Spinner>Spinning message</Spinner>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
