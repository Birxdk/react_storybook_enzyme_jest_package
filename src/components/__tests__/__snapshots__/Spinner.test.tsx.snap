// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wait spinner renders the spinner with children 1`] = `
.c1 {
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  -webkit-animation: spin 1s infinite;
  animation: spin 1s infinite;
}

.c0 {
  height: 100%;
  width: 100%;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .spinner-children {
  display: block;
  color: #000000;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="spinner-children"
    >
      Spinning message
    </div>
  </div>
</div>
`;

exports[`Wait spinner renders the spinner without children 1`] = `
.c1 {
  -webkit-transform-origin: center center;
  -ms-transform-origin: center center;
  transform-origin: center center;
  -webkit-animation: spin 1s infinite;
  animation: spin 1s infinite;
}

.c0 {
  height: 100%;
  width: 100%;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .spinner-children {
  display: block;
  color: #000000;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
